internal/queries/queries_consumer.go

```diff
@@ -45,10 +45,17 @@ func GetPayloadByRequestId(db *gorm.DB, request_id string) (result models.Payloa
 }
 
 func UpsertPayloadByRequestId(db *gorm.DB, request_id string, payload models.Payloads) (tx *gorm.DB, payloadId uint) {
+	columnsToUpdate := []string{"inventory_id", "system_id"}
```

Comment:
It seems to me that we should also be checking these like account and org id. These could theoretically be updated with empty strings as well no?

Reply:
Sure, I can add that ... @tahmidefaz what's your opinion? I felt that account/org ID were the important "must haves" because when org ID is absent, then you are unable to use the ingress `/track` API. But if `inventory_id` and `system_id` should never be switched to `""` for a `request_id` I'll add the lines to check those fields too.

Reply:
I think we can keep the logic for `inventory_id` and `system_id` as it is right now. We can address the `inventory_id` and `system_id` logic separately if it becomes an issue. :thinking:

Reply:
@dehort what do you think?

Reply:
I think we should include this same check on the `inventory_id` and `system_id`.  It looks like without the check it would be possible for us to lose valid `inventory_id` and `system_id` data if services which are sending messages to payload-tracker that do not include those fields after a service has sent messages which do include those fields (timing is everything).  This would be difficult to track down.

@bsquizz thank you for tracking this down and for the PR!  If you do not have time to fix those other fields, just let us know and we can take it from here.

Reply:
@dehort just got back to this, updated it to check the other fields as well. If all 4 fields happen to be empty (shouldn't even happen... but just in case...), we "do nothing"